jobrunner/executors/local.py

```diff
@@ -395,9 +395,26 @@ def finalize_job(job_definition):
 
     # First get the user-friendly message for known database exit codes, for jobs
     # that have db access
-    message = None
+    unmatched_hint = None
 
-    if exit_code == 137 and job_definition.cancelled:
+    if exit_code == 0 and unmatched_patterns == [] and outputs is not None:
+        message = "Completed successfully"
+
+    elif (
+        exit_code == 0
+        and unmatched_patterns is not None
+        and unmatched_outputs is not None
```

Comment:
What about if there are unmatched patterns, but no unmatched outputs at all? I think we should still trigger this branch in that case?

Having unmatched outputs is not an error - only having unmatched patterns is an error, basically.